contrib/buildrpm/buildrpmLibfabric.sh

```diff
@@ -0,0 +1,332 @@
+# Check if tarball was specified
+################################
+tarball="$1"
+
+if [[ ! -r "$tarball" ]]; then
+  error 2 "Can't find $tarball"
+fi
+verbose "Found tarball: $tarball"
+
+##############################
+# Get tarball name and version
+##############################
+tardirname=$(basename "$tarball" | awk '{print substr ($0, 0, index($0, ".tar")-1)}')
+if [[ -z "$tardirname" ]]; then
+  error 3 "Cannot determine name from $tarball"
+fi
+verbose "Name of package: $tardirname"
+
+version=$(basename "$tardirname" | cut -d- -f2)
+if [[ -z "$tardirname" ]]; then
+  error 4 "Cannot determine version from $tarball"
+fi
+verbose "Version: $version"
+
+if [[ -n "$install_in_opt" ]]; then
+  prefix="/opt/libfabric/$version"
+  rpmbuild_options="$rpmbuild_options --define '_prefix $prefix'"
+  verbose "Setting RPM install path to: $prefix"
+fi
+
+######################################
+# Try to unpack spec file from tarball
+######################################
+if [[ -f "$specfile" ]]; then
+  if [[ -n "$verbose" ]]; then
+    echo "$0: --> WARNING: $specfile already exists and will be overwritten" 1>&2
+  fi
+fi
+
+if [[ -n "$unpack_spec" ]]; then
+  verbose "Extracting tarball"
+  runcmd "tar $verboseoption -xf \"$tarball\""
+  verbose "Copying specfile"
+  runcmd "cp -fp $verboseoption \"$tardirname/$specfile\" ."
+  verbose "Cleanup after extraction"
+  runcmd "rm -rf $verboseoption \"$tardirname\""
+fi
+
+##############################
+# Check if specfile is present
+##############################
+if [[ ! -r $specfile ]]; then
+  if [[ -z "$noop" ]]; then
+    error 5 "Cannot find $specfile"
+  else
+    verbose "WARNING: Cannot find $specfile"
+  fi
+else
+  if [[ -n "$verbose" ]]; then
+    echo "$0: --> Found specfile: $specfile"
+  fi
+fi
+
+#############################
+# Prepare directory for build
+#############################
+verbose "Cleanup old rpmbuild directory"
+runcmd "rm -rf $verboseoption \"$rpmbuilddir\""
+verbose "Create rpmbuild SOURCES directory"
+runcmd "mkdir -p $verboseoption \"$rpmbuilddir/SOURCES/\""
+verbose "Copy tarball to rpmbuild directory"
+runcmd "cp -fp $verboseoption \"$tarball\" \"$rpmbuilddir/SOURCES/\""
+
+###########
+# Build RPM
+###########
+cmd=""
+build_opt=""
+
+if [[ -z "$verbose" ]]; then
+  build_opt="--quiet"
+else
+  build_opt="-v"
+fi
+cmd="rpmbuild $build_opt -bb $specfile $rpmbuild_options \
+  --define '_topdir $rpmbuilddir' \
+  --define '_sourcedir $rpmbuilddir/SOURCES' \
+  --define '_rpmdir $rpmbuilddir/RPMS' \
+  --define '_specdir $rpmbuilddir/SPECS' \
+  --define '_tmppath $rpmbuilddir/tmp'"
+
+if [[ -n "$configure_options" ]]; then
+  cmd="$cmd --define 'configopts $configure_options'"
+fi
+
+verbose "Build command used:"
+verbose "$cmd"
+verbose "RPM build will start"
+
+if [[ -z "$noop" ]]; then
+  eval "$cmd"
+
+  status="$?"
+  if [[ "$status" != "0" ]]; then
+    echo "$0: FATAL: *** FAILURE BUILDING RPM! ERROR CODE: $status" 1>&2
+    error 6 "Aborting"
+  fi
```

Comment:
By not using `runcmd` here:

1. You had to add an extra path to handle the dry run case (vs. using the infrastructure provided by `dryrun`)
1. You duplicate code

If *any* of the commands executed by `runcmd` fail, the script aborts (because *all* the commands are important).  I.e., saying that this command is more important than the rest isn't quite correct.

Having this command abort differently than the rest of the script is... a little weird.